internal/os/greenboot_scripts.go

```diff
@@ -12,40 +12,36 @@ echo "" >> /var/roothome/greenboot.log`
 
 	GreenbootHealthCheckScript = `#!/bin/bash
 
-if [ -x /usr/libexec/yggdrasil/device-worker ]; then
-echo "device-worker found, check passed!"
-else
-echo "device-worker not found, check failed!"
-exit 1
-fi
-
-if [ "$(systemctl is-active yggdrasild.service)" = "active" ]; then
-echo "yggdrasild.service is active, check passed!"
-else
-echo "yggdrasild.service is not active, check failed!"
-exit 1
-fi
+#!/bin/bash
 
-if [ "$(systemctl is-active nftables.service)" = "active" ]; then
-echo "nftables.service is active, check passed!"
-else
-echo "nftables.service is not active, check failed!"
-exit 1
-fi
-
-if [ "$(systemctl is-active podman.service)" = "active" ]; then
-echo "podman.service is active, check passed!"
+if [ -x /usr/libexec/yggdrasil/device-worker ]; then
+  echo "device-worker found, check passed!"
 else
-echo "podman.service is not active, check failed!"
-exit 1
+  echo "device-worker not found, check failed!"
+  exit 1
 fi
 
-if [ "$(systemctl is-active podman.socket)" = "active" ]; then
-echo "podman.socket is active, check passed!"
-else
-echo "podman.socket is not active, check failed!"
-exit 1
-fi
+CheckServiceIsActive(){
```

Comment:
just in term of naming convention, pls change to check_is_service_active
also, for simplifying, instead of passing a similar text as arguments, just the name of the service is enough.
so replace a call such as:
```
CheckServiceIsActive "$(systemctl is-active yggdrasild.service)" "yggdrasild.service is active, check passed!" "yggdrasild.service is not active, retrying!" "yggdrasild"
```
with
```
check_is_service_active yggdrasild.service
check_is_service_active nftables.service
...
```
and  both the messages and the command to invoke will be created inside the function, since they are sharing the same format and logic.